### PR TITLE
kubernetes_cluster supports maintenance_config

### PR DESCRIPTION
**Pull request by ms-henglu**
rewrite https://github.com/terraform-providers/terraform-provider-azurerm/pull/12305, embed maintenance resource to kubernetes_cluster
```
=== RUN   TestAccKubernetesCluster_updateMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_updateMaintenanceConfig
=== CONT  TestAccKubernetesCluster_updateMaintenanceConfig
--- PASS: TestAccKubernetesCluster_updateMaintenanceConfig (913.75s)

=== RUN   TestAccKubernetesCluster_basicMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_basicMaintenanceConfig
=== CONT  TestAccKubernetesCluster_basicMaintenanceConfig
--- PASS: TestAccKubernetesCluster_basicMaintenanceConfig (910.13s)

=== RUN   TestAccKubernetesCluster_completeMaintenanceConfig
=== PAUSE TestAccKubernetesCluster_completeMaintenanceConfig
=== CONT  TestAccKubernetesCluster_completeMaintenanceConfig
--- PASS: TestAccKubernetesCluster_completeMaintenanceConfig(916.23s)
```

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
THis is causing all tests to fail:
```
------- Stdout: -------
=== RUN   TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
=== PAUSE TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
=== CONT  TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain
    testcase.go:88: Step 1/1 error: Error running apply: exit status 1
        
        Error: retrieving Maintenance Configuration for Managed Kubernetes Cluster ("Cluster: (Managed Cluster Name \"acctestaks210730050729703273\" / Resource Group \"acctestRG-aks-210730050729703273\")"): containerservice.MaintenanceConfigurationsClient#Get: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="NotFound" Message="Could not find the maintenance configuration: default in subscription: *******, resourceGroup: acctestRG-aks-210730050729703273, resourceName: acctestaks210730050729703273."
        
          with azurerm_kubernetes_cluster.test,
          on terraform_plugin_test.tf line 28, in resource "azurerm_kubernetes_cluster" "test":
          28: resource "azurerm_kubernetes_cluster" "test" {
        
--- FAIL: TestAccKubernetesCluster_privateClusterOnWithPrivateDNSZoneSubDomain (1021.31s)
FAIL
```

**Comment by ms-henglu** [comment]
@katbyte , Hi, I've updated the commit, could you please take another look? Thanks!

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 1]
```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"maintenance_config": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:335]: i think config is redundant here, this is more a window?
```suggestion
			"maintenance_window": {
```

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 2]
```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
...
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"maintenance_allowed": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:341]: and maintaince is redundant
```suggestion
						"allowed": {
```

File: azurerm/internal/services/containers/kubernetes_cluster_resource.go [hunk 3]
```diff
@@ -331,6 +332,71 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
...
+											string(containerservice.WeekDayMonday),
+											string(containerservice.WeekDayTuesday),
+											string(containerservice.WeekDayWednesday),
+											string(containerservice.WeekDayThursday),
+											string(containerservice.WeekDayFriday),
+											string(containerservice.WeekDaySaturday),
+										}, false),
+									},
+
+									"hour_slots": {
```
Comment (katbyte) [inline azurerm/internal/services/containers/kubernetes_cluster_resource.go:361]: does slots make sense? hours seems fine
```suggestion
									"hours": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
@katbyte , thanks! I've added a commit to rename the fields according to your suggestions.

**Review by katbyte — APPROVED** [review]
thanks @ms-henglu - LGTM 🚀

**Pull request merged**